lib/src/entrypoint.dart

```diff
@@ -352,22 +368,29 @@ class Entrypoint {
     final newLockFile = await result.downloadCachedPackages(cache);
 
     final report = SolveReport(
-        type, root, lockFile, newLockFile, result.availableVersions, cache,
-        dryRun: dryRun);
-    if (!onlyReportSuccessOrFailure) {
-      await report.show();
+      type,
+      root,
+      lockFile,
+      newLockFile,
+      result.availableVersions,
+      cache,
+      dryRun: dryRun,
+      enforceLockfile: enforceLockfile,
+      quiet: onlyReportSuccessOrFailure,
+    );
+
+    final hasChanges = await report.show();
+    await report.summarize();
+    if (enforceLockfile && hasChanges) {
+      dataError(
+          'Could not enforce the lockfile$suffix.${onlyReportSuccessOrFailure ? forDetails() : ''}');
```

Comment:
I'm confused, shouldn't this say:
> Cannot not enforce `pubspec.lock`, try `dart pub get` without `--enforce-lockfile` to resolve dependencies.

Reply:
The discrepancies will have been output by report.show+summarize at this point.

But adding the suggestions to try without `--enforce-lockfile` is good.